src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -256,26 +253,48 @@ public void reset() {
         String d = serverContext.getDataStore().getLogDir();
         localLog.close();
         if (d != null) {
-            Path dir = FileSystems.getDefault().getPath(d);
-            String prefixes[] = new String[]{"log"};
-
-            for (String pfx : prefixes) {
-                try (DirectoryStream<Path> stream =
-                             Files.newDirectoryStream(dir, pfx + "*")) {
-                    for (Path entry : stream) {
-                        // System.out.println("Deleting " + entry);
-                        Files.delete(entry);
-                    }
-                } catch (IOException e) {
-                    log.error("reset: error deleting prefix " + pfx + ": " + e.toString());
+            Path dir1 = FileSystems.getDefault().getPath(d + "/log");
+            String report = "";
+
+            // Today I learned that the globbing done by newDirectoryStream does not
+            // handle any subdirectory "*/*" matching.  :-(
+            // Use try-with-resources to avoid leaking file descriptor on the directory.
+            try (DirectoryStream<Path> stream = Files.newDirectoryStream(dir1, "*-*-*-*-*")) {
+                for (Path entry : stream) {
+                    report = entry.toString();
+                    log.trace("Deleting recursively " + report);
+                    delete_all_matching_files(entry, "*");
+                    Files.delete(entry);
                 }
+            } catch(IOException e){
+                log.error("reset: error deleting prefix " + report + ": " + e.toString());
+            }
+            try {
+                delete_all_matching_files(dir1, "*");
+            } catch(IOException e){
+                log.error("reset: error deleting prefix " + dir1 + "/*: " + e.toString());
             }
         }
         reboot();
     }
 
+    private void delete_all_matching_files(Path dir, String glob) throws IOException {
```

Comment:
Please conform to method naming used by the rest of this project, which currently follows the Sun code conventions document.

This means that you should rename this method 
`deleteAllMatchingFiles`.